src/adventure/adventuretracker.cpp

```diff
@@ -0,0 +1,179 @@
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include "adventuretracker.h"
+#include "parser/parserutils.h"
+
+#include <QDebug>
+
+AdventureTracker::AdventureTracker(GameObserver &observer, QObject *const parent)
+    : QObject{parent}
+    , m_observer{observer}
+    , m_session{nullptr}
+{
+    connect(&m_observer,
+            &GameObserver::sig_sentToUserText,
+            this,
+            &AdventureTracker::slot_onUserText);
+
+    connect(&m_observer,
+            &GameObserver::sig_sentToUserGmcp,
+            this,
+            &AdventureTracker::slot_onUserGmcp);
+}
+
+void AdventureTracker::slot_onUserText(const QByteArray &ba)
+{
+    QString line = QString::fromLatin1(ba).trimmed();
```

Comment:
We might be doing this operation in a ton of files. Maybe we need another version of this signal that has it?